pkg/pkg.bzl

```diff
@@ -31,15 +33,32 @@ def _quote(filename, protect = "="):
     """Quote the filename, by escaping = by \\= and \\ by \\\\"""
     return filename.replace("\\", "\\\\").replace(protect, "\\" + protect)
 
+
 def _pkg_tar_impl(ctx):
     """Implementation of the pkg_tar rule."""
 
     # Files needed by rule implementation at runtime
     files = []
 
+    if ctx.attr.package_file_name:
+        output_name = ctx.attr.package_file_name
+        package_variables = ctx.attr.package_variables[PackageVariablesInfo]
+        if package_variables:
+            output_name = output_name.format(**package_variables.values)
+        elif ctx.attr.package_file_name.find('{') >= 0:
+            fail('attribute package_file_name requires package_variables')
```

Comment:
I can't check first, because there is no exeception handling. It fails hard if the variable is not there.

Can I rant about Starlark's mixed opinion on variable substitution. For strings, there is only string format style 
 {var} (there is makevars substitution, but that is deprecated).  For templates from files, there is only Make style $(var).    
1. this is inconsistent
2. neither is good. Starlark borrows heavily from Python. We should just use python string template syntax for both.

Reply:
LGTM.